### PR TITLE
feat: notion connector

### PR DESCRIPTION
**Pull request by shrikantNemiwal1**
## Description

Implemented notion connector

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Is there an addition/change in API Request, Response? If yes, documentation is required)
- [ ] Performance improvement
- [ ] Code refactoring
- [ ] Security fix

## Related Issues

- Fixes #[issue number]
- Closes #[issue number]
- Related to #[issue number]

## How Has This Been Tested?

[Describe the tests you ran to verify your changes. Provide instructions so we can reproduce.]

### Test Configuration

- [ ] Unit tests pass
- [ ] Integration tests pass
- [ ] Manual testing completed
- [ ] Cross-browser testing (if applicable)
- [ ] Mobile responsiveness tested (if applicable)

## Core Functionality Testing

Please confirm that the following core functionalities are working as expected:

- [ ] **Search capabilities** are working correctly
- [ ] **Knowledge search** is functioning properly
- [ ] **Connector indexing** is working as expected
- [ ] **Citations** are displaying and linking correctly
- [ ] **Documentation** is updated at https://docs.pipeshub.com/introduction

## What You Have Tested

Please describe what you have specifically tested:

- [ ] Feature works in development environment
- [ ] Feature works in staging environment
- [ ] Error handling scenarios tested
- [ ] Edge cases considered and tested
- [ ] Performance impact assessed
- [ ] Security implications reviewed

### Test Results

[Provide details of your test results here]

## Screenshots/Videos

**Before:**
[Attach screenshots/videos showing the current behavior]

**After:**
[Attach screenshots/videos showing the new behavior]

**Note:** Screenshots are required before merge for UI changes.

## Code Quality Checklist

- [ ] My code follows the project's coding standards
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Documentation

- [ ] Documentation has been updated (if applicable)
- [ ] API documentation updated (if applicable)
- [ ] README updated (if applicable)
- [ ] Changelog updated (if applicable)

## Security Considerations

- [ ] No sensitive data is exposed
- [ ] Input validation is implemented where necessary
- [ ] Authentication/authorization is properly handled
- [ ] No security vulnerabilities introduced

## Breaking Changes

- [ ] This PR introduces breaking changes
- [ ] Migration guide provided (if applicable)
- [ ] Version bump required

If breaking changes are introduced, please describe them here:

[Describe any breaking changes and how users should adapt]

## Performance Impact

- [ ] No performance impact
- [ ] Performance improved
- [ ] Performance impact assessed and acceptable

[Describe any performance implications]

## Dependencies

- [ ] No new dependencies added
- [ ] New dependencies are necessary and approved
- [ ] Dependencies updated and tested

List any new dependencies:
- [Dependency name and version]

## Deployment Notes

[Any special deployment considerations or steps]

## Checklist Before Merge

- [ ] All tests are passing
- [ ] Code review completed and approved
- [ ] Documentation updated and reviewed
- [ ] Screenshots/videos attached for UI changes
- [ ] Core functionality verified
- [ ] Security review completed (if applicable)
- [ ] Performance impact assessed
- [ ] Ready for production deployment

## Additional Notes

[Any additional information that would be helpful for reviewers]

---

**For Reviewers:**

Please ensure all checklist items are completed before approving this PR. Pay special attention to:
- Core functionality testing
- Security implications
- Performance impact
- Documentation completeness

### CODE REVIEW

File: backend/python/app/connectors/core/base/token_service/oauth_service.py [hunk 1]
```diff
@@ -206,15 +235,43 @@ async def exchange_code_for_token(self, code: str, state: Optional[str] = None,
...
+        # Prepare POST request kwargs (JSON or form-encoded)
+        post_kwargs = {"headers": headers}
+        if use_json_body:
+            headers["Content-Type"] = "application/json"
+            post_kwargs["json"] = data
+        else:
+            post_kwargs["data"] = data
+
+        # Make token refresh request
+        async with session.post(self.config.token_url, **post_kwargs) as response:
```
[the remark this answers is not in the document]
Comment (shrikantNemiwal1) [reply backend/python/app/connectors/core/base/token_service/oauth_service.py]: fixed

File: backend/python/app/connectors/core/base/token_service/oauth_service.py [hunk 2]
```diff
@@ -157,50 +157,71 @@ def _get_authorization_url(self,state: str, **kwargs) -> str:
...
                 masked_client_id = self.config.client_id[:FIRST_8_CHARS] + "..." if len(self.config.client_id) > FIRST_8_CHARS else "***"
                 error_msg = (
-                    f"OAuth token exchange failed with status {response.status}. "
+                    f"OAuth token request failed with status {response.status}. "
                     f"Token URL: {self.config.token_url}, "
                     f"Redirect URI: {self.config.redirect_uri}, "
                     f"Client ID (masked): {masked_client_id}, "
                     f"Response: {error_text}"
                 )
                 raise Exception(error_msg)
```
[the remark this answers is not in the document]
Comment (shrikantNemiwal1) [reply backend/python/app/connectors/core/base/token_service/oauth_service.py]: fixed

File: backend/python/app/connectors/core/base/token_service/oauth_service.py [hunk 3]
```diff
@@ -209,18 +230,16 @@ async def refresh_access_token(self, refresh_token: str) -> OAuthToken:
...
-            response.raise_for_status()
-            token_data = await response.json()
+        try:
+            token_data = await self._make_token_request(data)
+        except Exception as e:
+            # Enhance error message for 403 errors (common with expired/invalid refresh tokens)
+            error_str = str(e)
+            if "403" in error_str or "Forbidden" in error_str:
+                raise Exception(f"Token refresh failed with 403 Forbidden. This usually means the refresh token has expired or is invalid. {error_str}")
+            raise
```
[the remark this answers is not in the document]
Comment (shrikantNemiwal1) [reply backend/python/app/connectors/core/base/token_service/oauth_service.py:245]: fixed

File: backend/python/app/events/processor.py [hunk 4]
```diff
@@ -600,6 +599,174 @@ async def process_docx_document(
...
+                                        else:
+                                            row_dicts.append({})
+
+                    # Generate LLM row descriptions (skip header row)
+                    if row_dicts and len(row_dicts) > 1:
+                        try:
+                            table_data = {"grid": [[row] for row in row_dicts[1:]]}  # Skip header
+                            row_descriptions, _ = await get_rows_text(
+                                self.config_service, table_data, table_summary, column_headers
+                            )
```
[the remark this answers is not in the document]
Comment (shrikantNemiwal1) [reply backend/python/app/events/processor.py]: fixed

**Review by abhishek-pipeshub — APPROVED** [review]
